crates/outbound-http/src/host_impl.rs

```diff
@@ -38,8 +39,28 @@ impl OutboundHttp {
 
 #[async_trait]
 impl outbound_http::Host for OutboundHttp {
+    #[instrument(name = "spin_outbound_http.send_request", skip(self, req), err(level = Level::INFO),
+        fields(otel.kind = "client", url.full = Empty, http.request.method = Empty,
+        http.response.status_code = Empty, otel.name = Empty, server.address = Empty, server.port = Empty))]
     async fn send_request(&mut self, req: Request) -> Result<Result<Response, HttpError>> {
         Ok(async {
+            let current_span = tracing::Span::current();
+            let method = format!("{:?}", req.method)
+                .strip_prefix("Method::")
+                .unwrap_or("_OTHER")
+                .to_uppercase();
+            current_span.record("otel.name", method.clone());
+            current_span.record("url.full", req.uri.clone());
+            current_span.record("http.request.method", method);
+            if let Ok(uri) = req.uri.parse::<Uri>() {
+                if let Some(authority) = uri.authority() {
+                    current_span.record("server.address", authority.host());
+                    if let Some(port) = authority.port() {
+                        current_span.record("server.port", port.as_u16());
+                    }
+                }
+            }
+
```

Comment:
Per other comments it would be nice to extract this at some point in the not-too-distant future.

Reply:
In this particular case I wouldn't use the attribute macro, I'd manage the span (more) manually